src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer<Machine> machineProducer;
+        if (type == Bulldozer.class) {
+            machineProducer = new BulldozerProducer();
+        } else if (type == Excavator.class) {
+            machineProducer = new ExcavatorProducer();
+        } else if (type == Truck.class) {
+            machineProducer = new TruckProducer();
+        } else {
+            return new ArrayList<Machine>();
+        }
+        List<? extends Machine> machines = machineProducer.get();
+        return new ArrayList<>(machines);
+    }
```

Comment:
```suggestion
    private MachineProducer<Bulldozer> bulldozerProducer = new BulldozerProducer();
    private MachineProducer<Excavator> excavatorProducer = new ExcavatorProducer();
    private MachineProducer<Truck> truckProducer = new TruckProducer();

    @Override
    public List<Machine> getAll(Class<? extends Machine> type) {
        if (type != null) {
            if (type == Bulldozer.class) {
                return new ArrayList<>(bulldozerProducer.get());
            }
            if (type == Excavator.class) {
                return new ArrayList<>(excavatorProducer.get());
            }
            if (type == Truck.class) {
                return new ArrayList<>(truckProducer.get());
            }
        }
        return new ArrayList<>();
    }
```
you can create producers once and invoke them